web/saml.go

```diff
@@ -126,6 +130,7 @@ func completeSaml(c *Context, w http.ResponseWriter, r *http.Request) {
 			})
 		}
 
+		c.LogAuditWithUserId(user.Id, "success")
```

Comment:
Can't this still fail on the next line, shouldn't this line go below that check.

Reply:
There are two calls to DoLogin. The first one is with the SAML-related functionality and the second one is the the more generic login using the authenticated user. My reading of the ticket is that it asks for tracing for the SAML part - "successful authentication " for "SAML login", that is why I put it before the second one (to correspond to the "fail" statement above) but I can see how it could be interpreted as tracing of the whole flow. Maybe some of the confusion comes from the first DoLogin function which prob. should be called differently.

I propose adding another LogAuditWithUserId call after the second DoLogin with the "success" message and change the message to the first DoLogin LogAuditWithUserId() call to smth. like "obtained user from response"

Reply:
Sounds good...